install.ps1

```diff
@@ -3,67 +3,65 @@
-$xhsPath = "${destdir}xhs.exe"
-if (Test-Path -Path $xhPath -PathType Leaf) {
-    "Removing previous installation of xh from $($destdir)"
+# Check if an older version of xh.exe (includes xhs.exe) exists in '$destdir', if yes, then delete it, if not then download latest zip to extract from
+$xhPath = "${destdir}\xh.exe"
+$xhsPath = "${destdir}\xhs.exe"
+if (Test-Path -Path $xhPath -PathType Leaf)
+{
+    Write-Output "Removing previous installation of xh from $destdir"
     Remove-Item -r -fo $xhPath
     Remove-Item -r -fo $xhsPath
 }
 
-# xh.exe extraction start.
+# Create dir for result of extraction
+New-Item -ItemType Directory -Path $destdir -Force | Out-Null
 
+# Decompress the zip file to the destination directory
 Add-Type -Assembly System.IO.Compression.FileSystem
-
 $zip = [IO.Compression.ZipFile]::OpenRead($zipfile)
 $entries = $zip.Entries | Where-Object { $_.FullName -like '*.exe' }
+$entries | ForEach-Object { [IO.Compression.ZipFileExtensions]::ExtractToFile($_, $destdir + "\" + $_.Name) }
 
-# Create dir for result of extraction.
-
-New-Item -ItemType Directory -Path $destdir -Force | Out-Null
-
-# Extraction.
-
-$entries | ForEach-Object { [IO.Compression.ZipFileExtensions]::ExtractToFile( $_, $destdir + $_.Name) }
-
-# Free the zipfile.
-
+# Free the zipfile
 $zip.Dispose()
-
 Remove-Item -Path $zipfile
 
-# Copy xh.exe as xhs.exe into bin.
-
+# Copy xh.exe as xhs.exe into bin
 Copy-Item $xhPath $xhsPath
 
-# Add to environment variables.
-
-$p = [System.Environment]::GetEnvironmentVariable('Path', [System.EnvironmentVariableTarget]::User)
-if (!$p.ToLower().Contains($destdir.ToLower())) {
-
-    # Path to "user"/bin.
-
-    Write-Output "Adding $destdir to your Path"
-	
-    $p += "$destdir"
-    [System.Environment]::SetEnvironmentVariable('Path', $p, [System.EnvironmentVariableTarget]::User)
-    $Env:Path = [System.Environment]::GetEnvironmentVariable('Path', [System.EnvironmentVariableTarget]::Machine) + ";" + $p
+# Get version from zip file name.
+$xhVersion = $($zipfilename.trim("xh-v -x86_64-pc-windows-msvc.zip") )
+
+# Inform user where the executables have been put
+Write-Output "xh v$( $xhVersion ) has been installed to:`n - $xhPath`n - $xhsPath"
+
+# Make sure destdir is in the path
+$userPath = [System.Environment]::GetEnvironmentVariable('Path', [System.EnvironmentVariableTarget]::User)
+$machinePath = [System.Environment]::GetEnvironmentVariable('Path', [System.EnvironmentVariableTarget]::Machine)
+
+# If userPath AND machinePath both do not contain bin, then add it to user path
+if (!($userPath -contains $destdir) -and !($machinePath -contains $destdir))
+{
+    Write-Output "PATH did not contain $destdir"
+    
+    # Update userPath
+    $userPath = $userPath.Trim(";") + ";$destdir"
+    
+    # Modify PATH for new windows
+    Write-Output "Adding $destdir to user PATH."
+	[System.Environment]::SetEnvironmentVariable('Path', $userPath, [System.EnvironmentVariableTarget]::User)
+	Write-Output "Done."
```

Comment:
I think the first and third lines are redundant and could be removed
```
PATH did not contain $destdir
Adding $destdir to user PATH.
Done.
```

Reply:
Done